### PR TITLE
[codex] Add d64 RMSNorm public-row proof

### PR DESCRIPTION
**Pull request by omarespejel**
## Summary

- add the first native Stwo proof for the d64 RMSNorm public-row surface
- bind proof acceptance to exact checked public rows by recomputing preprocessed/base commitment roots before verification
- check verifier-side row semantics: input commitment, normalization config commitment, RMS scale tree root, sum of squares, and `rms_q8 = isqrt(floor(sum_squares / 64))`
- record the remaining scalar sqrt/range AIR hardening lane in issue #356

## Why

This crosses from statement-bound zkAI target work into an actual native AIR proof for a transformer primitive slice. The scope is intentionally narrow: RMSNorm public rows only, not full d64 block proof, not privacy, and not AIR-native integer sqrt/range soundness yet.

## Hardening

- verifier recomputes expected Stwo commitment roots from checked public rows and rejects mismatched roots before proof acceptance
- verifier pins the expected PCS profile before root recomputation or `verify`, so proof-controlled PCS config cannot drive twiddle sizing or security downgrades
- proof bytes are bounded before JSON deserialization; malformed commitment vectors fail closed before indexing
- public-row integer semantics use signed-M31 bounds, checked arithmetic, exact integer sqrt, and explicit zero-average rejection
- deterministic negative tests cover row arithmetic drift, scalar drift, commitment drift, non-claim drift, public-row tampering after proof generation, proof-byte tampering, PCS config drift, short commitment vectors, oversized proofs, and overflow-shaped inputs
- oracle/differential checks: not applicable in this PR beyond deterministic public-row recomputation; issue #356 tracks the next AIR-native sqrt/range oracle surface
- resource-bound/untrusted-input review: covered by proof-size cap, PCS profile pinning, commitment-vector bounds, and checked arithmetic before field encoding
- Kani/formal-kernel impact: not added here; this PR uses executable regression tests and a scoped non-claim that scalar sqrt/range remains verifier-side, not AIR-native

## Non-Claims

- not a full d64 block proof
- not private witness privacy
- not projection, activation, SwiGLU, down-projection, or residual proof
- not AIR-native sqrt/range soundness for `rms_q8`
- not binding the full d64 `output_activation_commitment` from only RMSNorm-local rows

## Validation

- `cargo +nightly-2025-07-14 test d64_native_rmsnorm_public_row_proof --lib --features stwo-backend`
- `python3 scripts/paper/paper_preflight.py --repo-root .`
- `git diff --check`
- `just gate-fast`
- `just gate`

## Summary by CodeRabbit

* **New Features**
  * Added a native proof flow for bounded 64-row RMSNorm public-row proofs with end-to-end proving and verification.

* **Verification**
  * Verification enforces strict numeric and bounds checks, recomputes and validates commitments, pins expected PCS configuration, and rejects malformed or oversized proofs before processing.

* **Documentation**
  * Added design and engineering docs describing the proof surface, exact verified relations, verifier-side checks, and explicit scope/limitations.

* **Tests**
  * Added tests covering round-trip proving/verification, tampering, malformed inputs, and boundary cases.

### CODE REVIEW

File: src/stwo_backend/d64_native_rmsnorm_public_row_proof.rs [hunk 1]
```diff
@@ -0,0 +1,1030 @@
...
+    if stark_proof.commitments[1] != expected_roots[1] {
+        return Err(public_row_error(
+            "base row commitment does not match checked public rows",
+        ));
+    }
+    let channel = &mut Blake2sM31Channel::default();
+    let commitment_scheme = &mut CommitmentSchemeVerifier::<Blake2sM31MerkleChannel>::new(config);
+    commitment_scheme.commit(stark_proof.commitments[0], &sizes[0], channel);
+    commitment_scheme.commit(stark_proof.commitments[1], &sizes[1], channel);
+    Ok(verify(&[&component], channel, commitment_scheme, stark_proof).is_ok())
```
[the remark this answers is not in the document]
Comment (omarespejel) [reply src/stwo_backend/d64_native_rmsnorm_public_row_proof.rs:545]: Keeping the exact length check intentionally for this v1 proof format. This verifier is fail-closed on serialized proof-shape drift: extra commitment roots are not treated as forward-compatible because they change the committed proof shape and should require a proof-version bump plus a verifier update. The code now documents that choice directly above the check, and the positive/negative tests cover both short and extra commitment vectors.
